app/models/account/holding.rb

```diff
@@ -1,6 +1,42 @@
 class Account::Holding < ApplicationRecord
+  include Monetizable
+
+  monetize :amount
+
   belongs_to :account
   belongs_to :security
 
+  validates :amount, :qty, :price, :currency, presence: true
+
   scope :chronological, -> { order(:date) }
+  scope :current, -> { where(date: Date.current).order(amount: :desc) }
+  scope :for, ->(security) { where(security_id: security).order(:date) }
+
+  delegate :name, to: :security
+  delegate :symbol, to: :security
+
+  def weight
+    portfolio_value = account.holdings.current.sum(&:amount)
+    portfolio_value.zero? ? 1 : amount / portfolio_value * 100
+  end
+
+  # Basic approximation of cost-basis
+  def avg_cost
+    avg_cost = account.holdings.for(security).where("date <= ?", date).average(:price)
+    Money.new(avg_cost, currency)
+  end
```

Comment:
Very simplistic approximation of "cost basis" of a holding.

In the future, we'll introduce other methods such as:

```ruby
def weighted_avg_cost
end

def fifo_cost
end
```